src/gui/mzroll/tabledockwidget.cpp

```diff
@@ -1113,82 +1113,123 @@ void TableDockWidget::deleteGroup(PeakGroup *groupX) {
-                        auto childGroupIter =
-                            find_if(begin(parentGroup->children),
-                                    end(parentGroup->children),
-                                    [&](PeakGroup& g) {
-                                        return g.getName() == name;
-                                    });
-                        if (childGroupIter != end(parentGroup->children)) {
-                            auto& childGroup = *childGroupIter;
-                            child->setData(0,
-                                           Qt::UserRole,
-                                           QVariant::fromValue(&childGroup));
-                        }
-                    }
+
+            if (!parentGroup->deleteChild(group))
+                continue;
+
+            itemsToDelete.insert(item);
+
+            // once a child is deleted, the pointers storing the
+            // location of memory blocks of child `PeakGroup`
+            // objects, may no longer be valid, therefore we
+            // update them.
+            for (int i = 0; i < parentItem->childCount(); ++i) {
+                QTreeWidgetItem* child = parentItem->child(i);
+                if (!child)
+                    continue;
+
+                auto name = child->text(1).toStdString();
+                auto childGroupIter =
+                    find_if(begin(parentGroup->children),
+                            end(parentGroup->children),
+                            [&](PeakGroup& g) {
+                                return g.getName() == name;
+                            });
+                if (childGroupIter != end(parentGroup->children)) {
+                    auto& childGroup = *childGroupIter;
+                    child->setData(0,
+                                   Qt::UserRole,
+                                   QVariant::fromValue(&childGroup));
                 }
             }
         }
-        if (parentGroup != NULL) {
-        	if (childrenNum == parentGroup->childCount()) {
-            	deleteGroup(group);
-            	nextItem = treeWidget->itemBelow(item->parent());
-            }
-        }
-      }
     }
-  }
-  if (nextItem)
-    treeWidget->setCurrentItem(nextItem, 0);
-  _mainwindow->getEicWidget()->replotForced();
-  showSelectedGroup();
-  _mainwindow->getEicWidget()->addPeakPositions();
 
-  return;
+    if (!groupsToDelete.empty()) {
+        allgroups.erase(remove_if(begin(allgroups),
+                                  end(allgroups),
+                                  [groupsToDelete](PeakGroup& g) {
+                                      return groupsToDelete.count(&g) > 0;
+                                  }),
+                        end(allgroups));
+    }
+
+    // reconnect selection trigger
+    connect(treeWidget,
+            SIGNAL(itemSelectionChanged()),
+            this,
+            SLOT(showSelectedGroup()));
+
+    if(nextItem){
+        treeWidget->setCurrentItem(nextItem);
+    }
```

Comment:
Does this work? I deleted groups in many different ways but never got a "next selection" by default. No items were highlighted and the EIC still showed me the last selected group (which was now deleted).